src/Middleware/HttpLogging/src/HttpLoggingOptions.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.HttpLogging
+{
+    /// <summary>
+    /// Options for the <see cref="HttpLoggingMiddleware"/>
+    /// </summary>
+    public sealed class HttpLoggingOptions
+    {
+        /// <summary>
+        /// Fields to log for the Request and Response. Defaults to logging request and response properties and headers.
+        /// </summary>
+        public HttpLoggingFields LoggingFields { get; set; } = HttpLoggingFields.RequestPropertiesAndHeaders | HttpLoggingFields.ResponsePropertiesAndHeaders;
+
+        /// <summary>
+        /// Request header values that are allowed to be logged.
+        /// </summary>
+        /// <remarks>
+        /// If a request header is not present in the <see cref="AllowedRequestHeaders"/>,
+        /// the header name will be logged with a redacted value.
+        /// </remarks>
+        public HashSet<string> AllowedRequestHeaders { get; } = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            HeaderNames.Accept,
+            HeaderNames.AcceptEncoding,
+            HeaderNames.AcceptLanguage,
+            HeaderNames.Allow,
+            HeaderNames.Connection,
+            HeaderNames.ContentType,
+            HeaderNames.Host,
+            HeaderNames.UserAgent
+        };
+
+        /// <summary>
+        /// Response header values that are allowed to be logged.
+        /// </summary>
+        /// <remarks>
+        /// If a response header is not present in the <see cref="AllowedResponseHeaders"/>,
+        /// the header name will be logged with a redacted value.
+        /// </remarks>
+        public HashSet<string> AllowedResponseHeaders { get; } = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            HeaderNames.ContentType,
+            HeaderNames.Date,
+            HeaderNames.ETag,
+            HeaderNames.Server,
```

Comment:
While safe, these aren't interesting. Date will be the same as the log timestamp. ETag is an opaque value. Server is always the same.
```suggestion
```